pymc_experimental/distributions/timeseries.py

```diff
@@ -0,0 +1,214 @@
+            P = pm.Dirichlet("P", a=[1, 1, 1], size=(3,))
+            # The initial state probabilities should have size = n_states, or 3 in this case.
+            init = pm.Categorical.dist(p = np.full(3, 1 / 3))
+            markov_chain = pm.DiscreteMarkovChain("markov_chain", P=P, init_dist=init, steps=100)
+    """
+
+    rv_type = DiscreteMarkovChainRV
+
+    def __new__(cls, *args, steps=None, initval="prior", **kwargs):
+        # Subtract 1 step to account for x0 given, better match user expectation of
+        # len(markov_chain) = steps
+        if steps is not None:
+            steps -= 1
+
+        steps = get_support_shape_1d(
+            support_shape=steps,
+            shape=None,
+            dims=kwargs.get("dims", None),
+            observed=kwargs.get("observed", None),
+            support_shape_offset=1,
+        )
+
+        return super().__new__(cls, *args, steps=steps, **kwargs)
+
+    @classmethod
+    def dist(cls, P=None, logit_P=None, steps=None, init_dist=None, **kwargs):
+
+        shape = kwargs.get("shape", None)
+
+        steps = get_support_shape_1d(support_shape=steps, shape=shape, support_shape_offset=1)
+
+        batch_size = None
+        if shape is not None:
+            batch_size = shape[1:] if shape[0] == steps else shape
+
+        if steps is None:
+            raise ValueError("Must specify steps or shape parameter")
+        if P is None and logit_P is None:
+            raise ValueError("Must specify P or logit_P parameter")
+        if P is not None and logit_P is not None:
+            raise ValueError("Must specify only one of either P or logit_P parameter")
+
+        if logit_P is not None:
+            P = pm.math.softmax(logit_P, axis=1)
+
+        P = pt.as_tensor_variable(P)
+        steps = pt.as_tensor_variable(intX(steps))
+
+        if init_dist is not None:
+            if not isinstance(init_dist, TensorVariable) or not isinstance(
+                init_dist.owner.op, (RandomVariable, SymbolicRandomVariable)
+            ):
+                raise ValueError(
+                    f"Init dist must be a distribution created via the `.dist()` API, "
+                    f"got {type(init_dist)}"
+                )
+            check_dist_not_registered(init_dist)
+            if init_dist.owner.op.ndim_supp > 1:
+                raise ValueError(
+                    "Init distribution must have a scalar or vector support dimension, ",
+                    f"got ndim_supp={init_dist.owner.op.ndim_supp}.",
+                )
+        else:
+            warnings.warn(
+                "Initial distribution not specified, defaulting to "
+                "`Categorical.dist(p=pt.full((k_states, ), 1/k_states), shape=...)`. You can specify an init_dist "
+                "manually to suppress this warning.",
+                UserWarning,
+            )
+            k = P.shape[0]
+            init_dist = pm.Categorical.dist(p=pt.full((k,), 1 / k), shape=batch_size)
```

Comment:
You don't need to worry about batch shapes, just the core shape. You will resize it anyway in `rv_op`

```suggestion
            init_dist = pm.Categorical.dist(p=pt.full((k,), 1 / k))
```